### PR TITLE
ES5 var to let and const

### PR DESCRIPTION
**Pull request by tusharf5**
**What kind of change does this PR introduce?**

refactoring

**Summary**

There are some var references which can be replaced with ES6's const and lets.

**Does this PR introduce a breaking change?**

No

**Other information**

### CODE REVIEW

**Comment by TheLarkInn** [comment]
@sokra are we still being strict on these for perf?

File: lib/OptionsDefaulter.js [hunk 1]
```diff
@@ -6,7 +6,7 @@
 
 function getProperty(obj, name) {
 	name = name.split(".");
-	for(var i = 0; i < name.length - 1; i++) {
```
Comment (michael-ciniawsky) [inline lib/OptionsDefaulter.js]: While `node =< v4.0.0` is the target this needs to keep using `var` as there is a known scope bug for loops in the V8 version `node =< v4.0.0` uses

Comment (tusharf5) [reply lib/OptionsDefaulter.js]: Thanks for reviewing. Should I change let to var here only or everywhere in the for loops?

Comment (sokra) [reply lib/OptionsDefaulter.js]: @michael-ciniawsky which bug? Is this snippet affected by this bug?

Comment (michael-ciniawsky) [reply lib/OptionsDefaulter.js]: No sry this was a false alarm on my side this bug only seems to affects [`while` loops](https://github.com/nodejs/node/issues/10435)

A ~[5x perf hit](https://github.com/nodejs/node/issues/5815#issuecomment-199473180) it is to be expected , but that was already being mentioned :)

Comment (sokra) [reply lib/OptionsDefaulter.js]: You shouldn't use node 4 anyway when caring about performance... ^^

**Comment by sokra** [comment]
Thanks

**Pull request merged**